src/server/game/Scripting/ScriptDefines/CreatureScript.cpp

```diff
@@ -148,7 +148,6 @@ uint32 ScriptMgr::GetDialogStatus(Player* player, Creature* creature)
     ASSERT(creature);
 
     auto tempScript = ScriptRegistry<CreatureScript>::GetScriptById(creature->GetScriptId());
-    ClearGossipMenuFor(player);
```

Comment:
GossipMenu& gossipMenu = player->PlayerTalkClass->GetGossipMenu();
    if (gossipMenu.GetSenderGUID() == creature->GetGUID())
        ClearGossipMenuFor(player);

Reply:
What do you think of doing this？

Reply:
@AnchyDev @heyitsbench @lewpar

I was told this PR would be free? =D

Reply:
That's probably okay, if you wanna test it. You just don't want to clear if the GUID is not the same.